lib/letsencrypt.js

```diff
@@ -1,215 +1,104 @@
-                    leSettings.domains[d] = leSettings.domains[d].trim();
-                    // delete empty
-                    !leSettings.domains[d] && leSettings.domains.splice(d, 1);
-                }
-            }
+                // for lex outputs
+                console.debug = console.debug || console.log;
 
-            const lex = LE.create({
-                version:         'v02',
-                debug:           true,
-                configDir:       leDir,
-                agreeTos:        true,
-                store:           require('./letsencryptStore.js').create({debug: true, log: leLog}),
-                server:          'https://acme-v02.api.letsencrypt.org/directory', //'staging',
-                email:           leSettings.email,
-                approvedDomains: leSettings.domains,
-                log:             leLog
-            });
-
-            if (settings.leUpdate) {
-                settings.lePort = parseInt(settings.lePort, 10) || 80;
-                // handles acme-challenge and redirects to https
-                // used for validation of requests like  http://example.com/.well-known/acme-challenge/BLABALBAL
                 try {
-                    const server = require('http').createServer(lex.middleware(function redirectHttps(req, res) {
-                        res.setHeader('Location', `https://${req.headers.host}:${settings.port}${req.url}`);
-                        res.statusCode = 302;
-                        res.end('<!-- Hello Developer Person! Please use HTTPS instead -->');
-                    }));
-
-                    server.on('error', e => {
-                        if (e.toString().includes('EACCES') && settings.lePort <= 1024) {
-                            log.error(`node.js process has no rights to start LetsEncrypt challenge server on port ${settings.lePort}.\n` +
-                                `Do you know that on linux you need special permissions for ports under 1024?\n` +
-                                `You can call in shell following script to allow it for node.js: "iobroker fix"`
-                            );
-                        } else {
-                            log.error(`LetsEncrypt challenge server could not be started on port ${settings.lePort}: ${e}`);
-                        }
-                    });
-
-                    server.listen(settings.lePort, () => {
-                        log.info('LetsEncrypt challenge server is started on ' + settings.lePort);
-                    });
+                    if (!fs.existsSync(leDir)) {
+                        fs.mkdirSync(leDir);
+                    }
                 } catch (err) {
-                    log.error('LetsEncrypt challenge server could not be started: ' + err);
+                    log.error('Lets encrypt Directory does not exist and can not be created as ' + leDir + ': ' + err);
+                    return null;
```

Comment:
Shouldn't you resolve the promise?

Reply:
True. Or more useful, I think throw an error here.